src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -429,7 +429,7 @@ private Model GetVocabularyDictionary()
                         float temp;
                         string firstKey = wordsInFirstLine[0];
                         float[] firstValue = wordsInFirstLine.Skip(1).Select(x => float.TryParse(x, out temp) ? temp : Single.NaN).ToArray();
-                        if (!firstValue.Contains(Single.NaN))
+                        if (!firstValue.Contains(Single.NaN) && firstValue.Length == model.Dimension)
                             model.AddWordVector(ch, firstKey, firstValue);
                         else
                             ch.Warning($"Parsing error while reading model file: '{_modelFileNameWithPath}', line number 1");
```

Comment:
We should likely drop this warning. This is an expected case to hit for fastText models (which have a single-line header). No need for a warning each time a fastText model is used.

Reply:
i can remove the warning but is adding the dimension check here correct way to fix the attached issue ?

Reply:
Was this the regression?

https://github.com/dotnet/machinelearning/pull/545#pullrequestreview-139987084

It was in this exact same code on the initial PR and was doing dimension checking...

Reply:
Since the comment is collapsed by default, here is the screenshot

![image](https://user-images.githubusercontent.com/8291187/45375806-b087c600-b5bb-11e8-86ee-05a3fc9f84a2.png)

Reply:
Its regression in a sense that the code is working for TLC 3.10

Reply:
okay I just saw the first iteration of the code. removing that if statement resulted in always adding the header., I can revert the change and remove the warning